### DISCUSSION TITLE
Daikin EHSX08 - no value Flow sensor and Water pressure

### DISCUSSION THREAD

**Discussion by robertoP-git** [raomin/ESPAltherma#221] [Q&A]
Hi everybody.
I'm integrating the project with my Daikin EHSX08P50B heat pump.
in setup.h I selected -> #include "def/ALTHERMA(LT_DA_04-08KW).h"

I select
{0x62,9,105,2,-1,"Flow sensor (l/min)"},
{0x62,11,105,1,2,"Water pressure"}
but it returns no value

Is there anyone with the same heat pump that can help me? thank's
Roberto

**Comment by Tardymo** [comment]
These are some standard values, which should be present on all Altherma 3 models. Do you get any other values? What do you see in log (espaltherma/log topic)?

**Comment by xalex75** [comment]
unfortunately a couple of registries, and 0x62 is one of them, could not be read by espaltherma, even if the info are there and you can read them from the machine display.
I have the same problem.

**Reply by KalaNagHTD** [reply]
In the HPSU some values are not provided by the Daikin board A1P (via X10A) but by the Rotex board BM1 via CAN-bus. Important values are flow rate, water pressure and 3-way-valve positions. I read these values from CAN by using [pyHPSU](https://github.com/zanac/pyHPSU) which is installed on my RaspberryPi besides ESPAltherma.

**Reply by xalex75** [reply]
> In the HPSU some values are not provided by the Daikin board A1P (via X10A) but by the Rotex board BM1 via CAN-bus. Important values are flow rate, water pressure and 3-way-valve positions. I read these values from CAN by using [pyHPSU](https://github.com/zanac/pyHPSU) which is installed on my RaspberryPi besides ESPAltherma.

Here you are Emiliano!
I've read our articles many, many times, trying to understand how the 508 Compact works. :-)
So we need a dedicated PI and the PICAN2 module (not so cheap, but I'll look for it). I have a spare 3B+ that can be used for this purpose.
The missing values are needed for COP calculation, and they would be useful for other tasks and statistics too.

**Reply by KalaNagHTD** [reply]
You don‘t need a dedicated Pi! I have ESPAltherma and pyHPSU (with PICAN2) running on the sam Pi 3.

**Reply by KalaNagHTD** [reply]
The better link: [Spanni26/pyHPSU](https://github.com/Spanni26/pyHPSU)
The cheaper hardware: CAN HAT

**Reply by xalex75** [reply]
> You don‘t need a dedicated Pi! I have ESPAltherma and pyHPSU (with PICAN2) running on the sam Pi 3.

I'm running HA (HassOS) on a PI4, espaltherma is running on an ESP32 connected to HA by MQTT, and Smart Grid is managed by a D1 Mini connected with Esphome to HA. So I should use my spare PI3 for this project. But that's ok, I just need to buy the PICAN 2 board.

**Reply by KalaNagHTD** [reply]
A CAN HAT should work also and is cheaper. See my recent post.

**Comment by microjupiter** [comment]
Heads up,

you're emailing the wrong Kyle Wilson.
